raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1037,6 +1047,14 @@ private boolean handleFetchResponse(
                     logger.info("Truncated to offset {} from Fetch response from leader {}",
                         truncationOffset, quorum.leaderIdOrNil());
                 });
+            } else if (partitionResponse.snapshotId().epoch() >= 0 && partitionResponse.snapshotId().endOffset() >= 0) {
```

Comment:
Fix this logic to be more conservative. It should be a fatal error if one is negative and the other is positive.